src/finite_differencing.jl

```diff
@@ -1,20 +1,29 @@
 export check_Dv, compute_errs
 
 """
+    approximate_Dv(
+        f::Function,
+        ȳ::ArrayOr∇Real,
+        x::Tuple{Vararg{ArrayOr∇Real}},
+        v::Tuple{Vararg{ArrayOr∇Real}},
+    )
     approximate_Dv(f::Function, ȳ::ArrayOr∇Real, x::ArrayOr∇Real, v::ArrayOr∇Real)
 
-Estimate the directional derivative of `f` at `x` in direction `v`.
+Estimate the directional derivative of `f` at `x` in direction `v`. If the function has
+multiple arguments, `x` and `v` should be `Tuple`s of inputs and directions respectively.
 """
-function approximate_Dv(f::Function, ȳ::ArrayOr∇Real, x::ArrayOr∇Real, v::ArrayOr∇Real)
-    y1, y2 = f.((x - v, x + v))
+function approximate_Dv(
+    f::Function,
+    ȳ::ArrayOr∇Real,
+    x::Tuple{Vararg{ArrayOr∇Real}},
+    v::Tuple{Vararg{ArrayOr∇Real}},
+)
+    y1, y2 = f(map(-, x, v)...), f(map(+, x, v)...)
     length(y1) == length(ȳ) || throw(ArgumentError("length(y1) != length(y)."))
     return sum(ȳ .* (y2 - y1) / 2)
```

Comment:
We might want to consider using higher-order finite difference methods, e.g. https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta_methods